protobuf/plugin/customname.go

```diff
@@ -0,0 +1,50 @@
+package plugin
+
+import (
+	"strings"
+
+	"github.com/gogo/protobuf/gogoproto"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+	"github.com/gogo/protobuf/protoc-gen-gogo/generator"
+)
+
+// CustomName preprocess the field, and set the [(gogoproto.customname) = "..."]
+// if necessary, in order to avoid setting `gogoproto.customname` manually.
+// The automatically assigned name should conform to Golang convention.
+func CustomName(field *descriptor.FieldDescriptorProto) {
```

Comment:
Let's give this a name that corresponds to what it does. It doesn't just support custom names. It re-writes the customname for fields that are typically upper case `ID`. Something like `CustomNameID` or something would be better for the name of this plugin.